File: 0_GraphRepresentation_ADJACANCY_MATRIX.cpp
#include<iostream>
using namespace std;
int main()
{
    
    #ifndef ONLINE_JUDGE
    freopen("input.txt","r",stdin);
    freopen("output.txt","w",stdout);
    #endif
    

    int n,m;
    cin>>n>>m;
    int adj[n+1][n+1];
    for(int i=0;i<n+1;i++)
    {
        for(int j=0;j<n+1;j++)
        {
            adj[i][j]=0;
        }
    }
    for(int i=0;i<m;i++)
    {
        int u,v;
        cin>>u>>v;
        adj[u][v]=1;
        adj[v][u]=1; //this line is only for undirected graph
    }
    for(int i=0;i<n+1;i++)
    {
        for(int j=0;j<n+1;j++)
        {
            cout<<adj[i][j]<<" ";
        }cout<<endl;
    }
}

File: 0_GraphRepresentation_ADJACANCY_MATRIX.py
import sys
sys.stdin=open('input.txt','r')
sys.stdout=open('output.txt','w')

n,m=map(int,input().split()) # n = number of nodes, m = number of edges


adj=[]
for i in range(n+1):
    temp=[]
    for j in range(n+1):
        temp.append(0)
    adj.append(temp)

for row in adj:
    print(row)

for i in range(m):
    u,v=map(int,input().split())
    adj[u][v]=1
    adj[v][u]=1 # this line is only for undirected graph
print("Printing Graph: ")
for row in adj:
    print(row)